js/components/changesetNotice.js

```diff
@@ -151,24 +151,16 @@ const create_unified_changeset_notice = ({
             let rfc_select  = $(notice.refs.elem).find('select[name="reasons_for_change"]');
             let reasons_for_change = (rfc_select.val() || []).map(v => parseInt(v));
             let is_valid = false;
-            if (validation_type === 0 &&
-                (!rationale && rationale_required) ||
-                (reasons_for_change.length === 0 && rfc_required)) {
-                    is_valid = false;
+            switch (validation_type) {
+                case 1:
+                    // Uses RFC dropdown logic (either a rationale or a non-Other reason for change required):
+                    is_valid = (rationale || (reasons_for_change.length > 0 && reasons_for_change.indexOf(1) == -1))
+                    break;
+                case 0:
+                default:
+                    // Either a rationale is submitted, or there was no rationale form, or it was optional:
+                    is_valid = (rationale || !include_rationale || !rationale_required);
```

Comment:
I wonder if the default option needs to be a little more complex.  Should it check both the `rationale_required` and the `rfc_required` fields and their corresponding values in the form?

Reply:
Definitely needs to if we aren't going to remove rfc_required as an option.  I'm thinking maybe after we get everything moved over to the unified function we can wean ourselves off of a couple of those parameters (like, maybe RFC dropdown is complex enough to require the "validation_type" parameter and that simplifies some of it?)